### PR TITLE
feat: dynamic selectors with alert_foreign_logs_enricher

### PR DESCRIPTION
**Pull request by nherment**
This adds support for calling the logs enricher based on the labels of a prometheus alert

image

# Documentation
image
---------------------------
image
----------------------------------
image

### CODE REVIEW

File: playbooks/robusta_playbooks/alerts_integration.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 from hikaru.model.rel_1_26 import Node
 from kubernetes import client
 from kubernetes.client import V1Pod, V1PodList, exceptions
+from playbooks.robusta_playbooks.util import resolve_selectors
```
Comment (Avi-Robusta) [inline playbooks/robusta_playbooks/alerts_integration.py:11]: this import looks like the issue with pytest
```
2024-10-18 12:51:43.140 ERROR    failed to module alerts_integration
Traceback (most recent call last):
  File "/app/src/robusta/runner/config_loader.py", line 1[82](https://github.com/robusta-dev/robusta/actions/runs/11403920443/job/31731978176?pr=1600#step:11:83), in __import_playbooks_package
    m = importlib.reload(importlib.import_module(module_name))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 6[90](https://github.com/robusta-dev/robusta/actions/runs/11403920443/job/31731978176?pr=1600#step:11:91), in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/venv/lib/python3.11/site-packages/robusta_playbooks/alerts_integration.py", line 11, in <module>
    from playbooks.robusta_playbooks.util import resolve_selectors
ModuleNotFoundError: No module named 'playbooks'
```

File: playbooks/robusta_playbooks/util.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+import re
+from typing import Dict, List
+
+REGEXP_FIND_SELECTOR = r'\{\{\s*labels\.([^\s\}]+)\s*\}\}'
+
+def resolve_selectors(labels:Dict, selectors:List[str]) -> List[str]:
```
Comment (Avi-Robusta) [inline playbooks/robusta_playbooks/util.py:7]: I would add this above
`build_selector_query` in 
`/Users/avirobusta/git/robusta/src/robusta/integrations/kubernetes/custom_models.py`
and import from there

Comment (Avi-Robusta) [reply playbooks/robusta_playbooks/util.py:7]: look at `format_pod_templated_string`, you might be able to use it in a similar manner

File: playbooks/robusta_playbooks/alerts_integration.py (hunk 1 above)
Comment (arikalon1) [reply playbooks/robusta_playbooks/alerts_integration.py:11]: if it's a code that is common to different playbooks packages, maybe it's better to put in in the `core` package, and not in the `playbooks` package

playbooks packages are loaded dynamically, so having dependencies between playbooks file must validate the are loaded in the right order (with the `core` package, this problem doesn't exist)

File: playbooks/robusta_playbooks/util.py (hunk 2 above)
Comment (nherment) [reply playbooks/robusta_playbooks/util.py]: Done

File: playbooks/robusta_playbooks/alerts_integration.py (hunk 1 above)
Comment (nherment) [reply playbooks/robusta_playbooks/alerts_integration.py]: @arikalon1 As this PR adds no more imports than were already existing I suggest to make that change in another PR if you think it's warranted.

**Review by arikalon1 — APPROVED** [review]
nice work!

**Comment by nherment** [comment]
Fixes https://github.com/robusta-dev/robusta/issues/1529

**Pull request merged**